flock.go

```diff
@@ -39,6 +39,27 @@ func NewFlock(path string) *Flock {
 	return &Flock{path: path}
 }
 
+// Close will call Unlock() and close the underlying file descriptor. Use of this
+// function is not strictly necessary as the file descriptor will be cleaned up
+// by the garbage collector, but some long lived applications may not want to
+// depend on that.
+func (f *Flock) Close() error {
```

Comment:
I think we should keep this function, but maybe just have it invoke `return f.Unlock()`